docs: Russian translation

Alex-Sokolov opened the pull request:
Finally! :)

Some points:

It's convenient for me to save English localization files to make it easier to merge from upstream without conficts and later compare files side-by-side. That why russian locale stored in `/ru/` folder and not as default `/`

---

Links for page now contain doubled `ru`, for example `https://v3.ru.vuejs.org/ru/guide/introduction.html`. That's because vuepress didn't offer a way to pre-set default locale. Rewrite russian locale files as default locale is not useful (see above)

---

Root README.md file for english locale duplicated from russian. I don't know how to show /ru/README.md file if I need to show russian locale as default one.

---

Also had to duplicate translation here, because vuepress at start page show /README.md for english locale and not /ru/README.md for russian. And update popup also showed in english.

```
{
  serviceWorker: true,
  updatePopup: {
    '/': {
      message: 'Доступен новый контент.',
      buttonText: 'Обновить'
    },
    '/ru/': {
      message: 'Доступен новый контент.',
      buttonText: 'Обновить'
    }
  }
}
```

---

Vuepress show some pages with big empty space, but some don't (also codeblocks in headings don't have space around). See screenshots:

<img width="957" alt="Снимок экрана 2021-05-09 в 11 03 04" src="https://user-images.githubusercontent.com/4497128/117566896-ec9fbb80-b0c1-11eb-945f-a3c98b02652b.png">

<img width="1007" alt="Снимок экрана 2021-05-09 в 12 10 53" src="https://user-images.githubusercontent.com/4497128/117566816-81ee8000-b0c1-11eb-921f-bb4cd0b6dd86.png">

---

The name of the language for the code block overlaps with the content update message

<img width="942" alt="Снимок экрана 2021-05-09 в 13 50 02" src="https://user-images.githubusercontent.com/4497128/117569264-83be4080-b0cd-11eb-9e28-4f4d0ddca75f.png">

---

Also check with `vuepress-plugin-check-md` and some hashes showed as broken, but they exist and links works ok. Maybe also bug in plugin.

```
Hash is not found: [справочнике API](../api/global-api.md#аргументы-4) (translation-gang/docs-next/src/ru/guide/component-dynamic-async.md:92:43)

Hash is not found: [рекомендаций](../style-guide/#именование-базовых-компонентов-настоятельно-рекомендуется) (translation-gang/docs-next/src/ru/guide/component-registration.md:26:64)

Hash is not found: [рекомендаций](../style-guide/#избегаите-использования-v-if-с-v-for-важно) (translation-gang/docs-next/src/ru/guide/conditional.md:91:99)

Hash is not found: [вышеупомянутых](#vmodel-ime-tip) (translation-gang/docs-next/src/ru/guide/forms.md:259:86)

Hash is not found: [рекомендаций](../style-guide/#избегаите-использования-v-if-с-v-for-важно) (translation-gang/docs-next/src/ru/guide/list.md:244:115)

Hash is not found: [**должен** использоваться kebab-case](#стиль-именования-компонентов-в-шаблонах-настоятельно-рекомендуется) (translation-gang/docs-next/src/ru/style-guide/README.md:894:66)
```

---

Don't know where to start digging about this problems, but I don't have enough energy and time for that and if someone could help I will be glad for that.

skirtles-code reviewed (commented):
This looks great.

I can't review the translation itself as I don't know Russian but I've tried to provide general feedback.

---

Could you add `https://v3.ru.vuejs.org/` to [`vuejs/docs-next/src/.vuepress/components/guide/contributing/translations-data.js`](https://github.com/vuejs/docs-next/blob/master/src/.vuepress/components/guide/contributing/translations-data.js#L20) as part of this PR? The translation repo is already listed, it just needs the deployment URL.

---

On the front page, https://v3.ru.vuejs.org/, the green video button doesn't work. Clicking that button just navigates to https://v3.ru.vuejs.org/ru/ instead.

---

There are some links in the main navigation that link to the page https://v3.ru.vuejs.org/support-vuejs/. I believe they need a `/ru` adding to their path.

---

The search box is searching the English docs.

---

From https://v3.ru.vuejs.org/ru/cookbook/editable-svg-icons.html:

![image](https://user-images.githubusercontent.com/65301168/117600584-1dc0d000-b144-11eb-93cd-9436cd321e77.png)

I believe they're supposed to be bullet points rather than a quote.

---

Could you update the repo URL on GitHub?

![image](https://user-images.githubusercontent.com/65301168/117601086-47c6c200-b145-11eb-8011-87baf156b106.png)

---

For the root `README.md` (shown [here](https://github.com/translation-gang/docs-next#readme), not the one in `src`), could you update it with content relevant to that repo? That's the entry point for new contributors, so ideally it should confirm that they're in the right place. Explaining how you manage pulling in upstream changes might be worth a mention too.

Here are some examples from other translations if you're looking for inspiration:

* https://github.com/vuejs-id/docs-next/#readme
* https://github.com/vuejs/docs-next-zh-cn/#readme

---

The spacing problems you've observed with some headings in the migration guide are a problem in the English docs too. I think it's caused by the custom CSS that's used to insert the badges rather than VuePress itself. More investigation needed on our end.

---

The clash between the PWA popup and code blocks is being tracked here: https://github.com/vuejs/docs-next/issues/865

---

I wasn't aware of `vuepress-plugin-check-md`. I check the internal links in the docs using my own tool, https://github.com/skirtles-code/vue-docs-link-checker. I concur that you don't have any broken links.

Alex-Sokolov commented:
Thanks for review, @skirtles-code !

> Could you add https://v3.ru.vuejs.org/ to vuejs/docs-next/src/.vuepress/components/guide/contributing/translations-data.js as part of this PR

Added

> On the front page, https://v3.ru.vuejs.org/, the green video button doesn't work

Change url to `/#`, didn't find how to create action button without redirect in another way

> There are some links in the main navigation that link to the page https://v3.ru.vuejs.org/support-vuejs/.

Fixed

> I believe they're supposed to be bullet points rather than a quote.

Fixed

> Could you update the repo URL on GitHub?

Updated

> For the root README.md (shown here, not the one in src), could you update it with content relevant to that repo?

Updated

skirtles-code commented:
I think the broken search box is quite important.

![image](https://user-images.githubusercontent.com/65301168/117764279-25f03c80-b224-11eb-9d89-cac276bda0de.png)

It currently searches the hidden copy of the English docs. Clicking on a search result opens an English page without a sidebar.

I don't know how difficult it is to fix it. If it can't be configured to search the right locale then maybe just turn it off altogether for now?
